### DISCUSSION TITLE
Is it possible to use Isar without the generator?

### DISCUSSION THREAD

**Discussion by kaboc** [isar/isar#257] [Help] [answered]
It is possible in Hive to write everything manually, which is one of the things I like about Hive, and I'd like to use Isar in the same way. Is it  possible in the first place and documented anywhere about it?

**Comment by simc** [comment, marked as answer]
No that's not possible. Is there a specific reason you want to avoid the generator?

**Reply by kaboc** [reply]
Thanks for the answer, Simon.

Part of my dislike of code generation is due to bad experiences with `build_runner`. It's slow. I have to run it and wait for a while every time I change models. Additionally, it's a hassle to have to learn and remember how to use it. For example, when I'm assigned to a project that uses Freezed, which I personally don't like and avoid using, I have to open its document to look through so many features or see the usage of some unfamiliar notation like `_$` that I already learnt before but is a little hard to remember. Tools like that tend to bring another inconvenience in exchange for removing some. I even have to fix issues that Freezed causes occasionally. I'd rather do manually instead.

Compared to it, however, code generation for Isar seems to be very simple. If my understanding is right, what I need to learn is just a couple of annotations, apparently easy to understand, so I'll give it a try anyway.

If only some other generator (not dependent on build_runner) that gets the job done in an instant..., like, I change a model, hit Ctrl/Command + S, automatically triggering the generation, and it's already done in the next moment. Such a new tool would be great.

**Reply by simc** [reply]
Dart will soon have macros which resolve all the issues you describe.
The Isar annotations are designed in a way to allow us to switch to macros without you needing to change anything.

**Reply by kaboc** [reply]
Awesome! I hope it'll happen before long.
Now that It's been confirmed all the hassle will be gone in the future, I think I can start using Isar. Thank you!

**Reply by tomasweigenast** [reply]
> No that's not possible. Is there a specific reason you want to avoid the generator?

I add someting to @kaboc 
Avoiding using the generator is useful if you have a repository pattern, where your models are defined at "programming time". I mean, If I develop a repository pattern and I publish a package, I would like to use Isar as a store provider, but i don't want users have to design models for each provider implementation.

**Reply by simc** [reply]
@tomasweigenast I'm not exactly sure what you mean. Could you maybe give an example? You can always define a generic model and just store your data as serialized json 🤷🏼‍♂️
